rasa/nlu/utils/spacy_utils.py

```diff
@@ -129,18 +130,61 @@ def get_text(self, example, attribute):
 
         return self.preprocess_text(example.get(attribute))
 
+    @staticmethod
+    def merge_content_lists(freezed_indices, doc_lists):
+        dct = dict(freezed_indices)
+        dct.update(dict(doc_lists))
+        return list(dct.items())
+
     def docs_for_training_data(
         self, training_data: TrainingData
     ) -> Dict[Text, List[Any]]:
 
         attribute_docs = {}
         for attribute in SPACY_FEATURIZABLE_ATTRIBUTES:
-
             texts = [self.get_text(e, attribute) for e in training_data.intent_examples]
+            # Index and freeze indices of the training samples for preserving the order
+            # after processing the data.
+            freezed_indices = [(idx, text) for idx, text in enumerate(texts)]
+            # Currently, none existent attribute samples are None which leads to an error
+            # in spaCy and are therefore set as empty strings by Rasa. Since third-party
+            # libraries relying on spaCy might have problems with Doc-objects created on empty
+            # strings, we further treat them separately.
+            docs_to_pipe = list(filter(lambda fi_tpl: fi_tpl[1] != "", freezed_indices))
```

Comment:
Can you expand `fi_tpl` to a more meaningful name. In general, we try to avoid using short forms in variable names.